t3.c

```diff
@@ -0,0 +1,348 @@
+    }
+
+
+    int zero(double x){
+    const double eps = 1e-9;
+    return fabs(x) < eps;
+    }*/
+
+    printf("\nTask 6: Comparison by module\n");
+    printf("Enter three numbers: ");
+    scanf("%d %d %d", &a, &b, &c);
+
+    a = abs(a);
+    b = abs(b);
+    c = abs(c);
+
+    int max = a > b ? (a > c ? a : c) : (b > c ? b : c);
+    printf("Maximum by module: %d\n", max);
+}
+
+// Завдання 7(a): Розв'язання квадратного рівняння
+void task7_a() {
+    double a, b, c;
+
+    printf("\nExercise 7(a): Solving a quadratic equation\n");
+    printf("Enter the coefficients a, b, c for the equation ax^2 + bx + c = 0:\n");
+    printf("a = ");
+    scanf("%lf", &a);
+    printf("b = ");
+    scanf("%lf", &b);
+    printf("c = ");
+    scanf("%lf", &c);
+
+    /*if (zero(a) && !zero(b) && !zero(c)){
+    double x = - c / b;
+    printf("x1 = %lf \n", x);
+    }
+    if (zero(c) && !zero(a) && !zero(b)){
+    int x1 = 0;
+    double x2 = - a / b;
+    printf("x1 = %d, x2 = %lf \n", x1, x2);
+    }
+    if (zero(b) && !zero(a) && !zero(c)){
+    int x = - c / a;
+    if (x > 0){
+
+    double x1 = sqrt(x);
+    double x2 = - sqrt(x);
+    printf("x1 = %lf, x2 = %lf \n", x1, x2);
+    }
+
+    if (x < 0){
+    printf("No real roots.\n");
+    }
+    }
+    if (a > 0 && b > 0 && c >0){
+    int d = pow(b, 2) - 4 * a * c;
+    if (d > 0){
+    double x1 = ( -b + sqrt(d)) / (2 * a);
+    double x2 = ( -b - sqrt(d)) / (2 * a);
+    printf("x1 = %lf, x2 = %lf \n", x1, x2);
+    }
+    else if (d == 0){
+    double x = -b / (2 * a);
+    printf("x1 = %lf \n", x);
+    }
+    else if (d < 0){
+    printf("No real roots.\n");
+    }
+    }
+    if (zero(b) && zero(c) && !zero(a)){
+    if (a > 0){
+    double x1 = sqrt(a);
+    double x2 = - sqrt(a);
+    printf("x1 = %lf, x2 = %lf \n", x1, x2);
+    }
+    if (a < 0){
+    printf("No real roots\n");
+    }
+    }
+    if (zero(a) && zero(c) && !zero(b)){
+    double x1 = -b;
+    printf("x1 = %lf \n", x1);
+    }
+    if (zero(a) && zero(b)){
+    printf("No equation exists\n");
+    }*/
+
+    if (a == 0) {
+        printf("This is not a quadratic equation!\n");
+    } else {
+        double D = b * b - 4 * a * c;
+        if (D > 0) {
+            double x1 = (-b + sqrt(D)) / (2 * a);
+            double x2 = (-b - sqrt(D)) / (2 * a);
+            printf("Junctions: x1 = %.2lf, x2 = %.2lf\n", x1, x2);
+        } else if (D == 0) {
+            double x = -b / (2 * a);
+            printf("Solution: x = %.2lf\n", x);
+        } else {
+            printf("There are no real roots\n");
+        }
+    }
+}
+
+// Завдання 7(b): Розв'язання рівняння четвертого ступеня
+void task7_b() {
+    double a, b, c;
+
+    printf("\nExercise 7(b): Solving a quadratic equation\n");
+    printf("Enter the coefficients a, b, c for the equation ax^4 + bx^2 + c = 0:\n");
+    printf("a = ");
+    scanf("%lf", &a);
+    printf("b = ");
+    scanf("%lf", &b);
+    printf("c = ");
+    scanf("%lf", &c);
+
+    if (a == 0) {

```

Comment:
comparison is bad

Reply:
I saw that the comparison in this task is bad, because of this I completely changed the course of solving this problem